src/rpm_ostree/cli_deploy.rs

```diff
@@ -19,27 +23,42 @@ lazy_static::lazy_static! {
 pub fn deploy_locked(release: Release, allow_downgrade: bool) -> Fallible<Release> {
     DEPLOY_ATTEMPTS.inc();
 
-    let result = invoke_cli(release, allow_downgrade);
+    let result = invoke_cli(release, allow_downgrade, false);
     if result.is_err() {
         DEPLOY_FAILURES.inc();
     }
 
     result
 }
 
+/// Register as the update driver by calling `rpm-ostree deploy --register-driver`
+/// with an empty string as the revision to indicate that this is a no-op deployment.
+pub fn deploy_register_driver() -> Fallible<()> {
+    let empty_release = rpm_ostree::Release {
+        version: String::from(""),
+        checksum: String::from(""),
+        age_index: None,
+    };
+    invoke_cli(empty_release, false, true)?;
+    Ok(())
+}
+
 /// CLI executor.
-fn invoke_cli(release: Release, allow_downgrade: bool) -> Fallible<Release> {
+fn invoke_cli(release: Release, allow_downgrade: bool, register_driver: bool) -> Fallible<Release> {
```

Comment:
It feels like this could be split into a `invoke_cli_deploy` plus a `invoke_cli_register`.

Reply:
Sure! I was thinking that they share mostly the same code, but I guess forcing them together like this does feel a bit unnatural.

Reply:
Yes, I got the same feeling while reading through. The fake `Release` item in particular is a good indicator that this arrangement could be re-calibrated.